meson_options.txt

```diff
@@ -113,6 +113,11 @@ option('accelerate',
        value: true,
        description: 'Enable Accelerate BLAS support')
 
+option('metal',
+       type: 'boolean',
+       value: true,
```

Comment:
If you make this default true, you need to make sure all the requirements are satisfied before adding the files (and dependencies). See how it is done for cuda.

Reply:
The `dependency()` check works for the requirements as well.

Reply:
Nice.